### PR TITLE
4651: update document validation for signedAt and signedJudgeName requirements

### PR DESCRIPTION
**Pull request by rachaelparris**

### CODE REVIEW

**Review by kkoskelin — APPROVED** [review]

File: shared/src/business/entities/Document.js [hunk 1]
```diff
@@ -391,8 +391,8 @@ joiValidationDecorator(
       .string()
       .when('draftState', {
         is: joi.exist().not(null),
-        otherwise: joi.when('documentType', {
-          is: joi.valid(...ORDER_TYPES.map(t => t.documentType)),
+        otherwise: joi.when('eventCode', {
```
Comment (mrinsin) [inline shared/src/business/entities/Document.js:394]: Are there any unit tests that could be written to cover this new branch of logic, or is it covered in tests that may currently fail/need to be updated?

Comment (mrinsin) [reply shared/src/business/entities/Document.js:394]: (speaking from entity validation rule change related experience lol)

Comment (rachaelparris) [reply shared/src/business/entities/Document.js:394]: It's covered by tests that I updated in `Document.test.js`

Comment (rachaelparris) [reply shared/src/business/entities/Document.js:394]: Yep it was already well-covered, I just updated `documentType` to `eventCode` to use the new list.

**Review by mrinsin — APPROVED** [review]

**Pull request merged**